src/serializer/ResidualHeapSerializer.js

```diff
@@ -695,6 +701,120 @@ export class ResidualHeapSerializer {
+    let objectProperties: Map<string, any> = val.properties;
+    let typeValue = this._getPropertyValue(objectProperties, "type");
+    let keyValue = this._getPropertyValue(objectProperties, "key");
+    let refValue = this._getPropertyValue(objectProperties, "ref");
+    let propsValue = this._getPropertyValue(objectProperties, "props");
+
+    invariant(typeValue !== null, "JSXElement type of null");
+
+    let identifier = convertExpressionToJSXIdentifier(this.serializeValue(typeValue));
+    let attributes = [];
+    let children = [];
+
+    if (keyValue !== null) {
+      let keyExpr = this.serializeValue(keyValue);
+      if (keyExpr.type !== "NullLiteral") {
+        attributes.push(convertKeyValueToJSXAttribute("key", keyExpr));
+      }
+    }
+
+    if (refValue !== null) {
+      let refExpr = this.serializeValue(refValue);
+      if (refExpr.type !== "NullLiteral") {
+        attributes.push(convertKeyValueToJSXAttribute("ref", refExpr));
+      }
+    }
+
+    if (propsValue instanceof ObjectValue) {
+      // the propsValue is visited to get the properties, but we don't emit it as the object
+      // is contained within a JSXOpeningElement
+      this.serializedValues.add(propsValue);
+      // have to case propsValue to ObjectValue or Flow complains that propsValues can be null/undefined
+      for (let [key, propertyBinding] of (propsValue: ObjectValue).properties) {
+        let desc = propertyBinding.descriptor;
+        if (desc === undefined) continue; // deleted
+
+        invariant(key !== "key" && key !== "ref", `"${key}" is a reserved prop name`);
+
+        if (key === "children" && desc.value !== undefined) {
+          let childrenValue = desc.value;
+          if (childrenValue instanceof ArrayValue) {
+            this.serializedValues.add(childrenValue);
+            let childrenLength = this._getPropertyValue(childrenValue.properties, "length");
+            let childrenLengthValue = 0;
+            if (childrenLength instanceof NumberValue) {
+              childrenLengthValue = childrenLength.value;
+              for (let i = 0; i < childrenLengthValue; i++) {
+                let child = this._getPropertyValue(childrenValue.properties, "" + i);
+                if (child instanceof Value) {
+                  children.push(this._serializeValueReactElementChild(child));
+                } else {
+                  this.logger.logError(val, `JSXElement "props.children[${i}]" failed to serialize due to a non-value`);
+                }
+              }
+              continue;
+            }
+          }
+          if (childrenValue instanceof Value) {
+            children.push(this._serializeValueReactElementChild(childrenValue));
+          } else {
+            this.logger.logError(val, `JSXElement "props.children" failed to serialize due to a non-value`);
+          }
+          continue;
+        }
+        if (desc.value instanceof Value) {
```

Comment:
What if it's not a Value? That indicates that we are dealing with an accessor property. Log an error that it's not supported?